packages/target-electron/src/desktop_settings.ts

```diff
@@ -57,14 +54,9 @@ class PersistentState extends EventEmitter {
   saveImmediate(): Promise<void> {
     log.info(`Saving state to ${appConfig.filePath}`)
     const copy = Object.assign({}, this.inner_state)
-    return new Promise((res, rej) => {
-      appConfig.write(copy, (err: any) => {
-        if (err) {
-          log.error('State save failed', err)
-          rej(err)
-        }
-        res(err)
-      })
+    return appConfig.write(copy).catch(err => {
+      log.error('State save failed', err)
+      return err
```

Comment:
```suggestion
      throw err
```

Otherwise the promise returned from `saveImmediate` never rejects.